### ISSUE TITLE
Error in *writeRaster* with overwrite = TRUE, file already exists

### ISSUE THREAD

**Issue by bryce-carson** [rspatial/terra#1539]
[This script, preaggregator.tar.gz](https://github.com/user-attachments/files/15778270/preaggregator.tar.gz), fails due to one two errors:

1. (in this issue) "file already exists" despite the argument `overwrite = TRUE`, or
2. (in #1538) an "external reference" error.

I am re-running the script, which takes a few minutes, and then I will update this issue and the related issue with further information and explicit quotations of the error messages.

**Comment by bryce-carson** [comment]
[This population.xlsx file is required by the script, and should live in a sibling directory named `misc`](https://github.com/user-attachments/files/15778362/population.xlsx).

**Comment by bryce-carson** [comment]
The directory structure is as diagrammed below.

```
[bryce@fedora]~/src/r/spatialEpisim% tree -d -L 1
.
├── gadm
├── misc
├── observeddata
├── preaggregated_data
├── R
├── rsconnect
├── scripts
├── seeddata
├── tif
└── www
```

**Comment by bryce-carson** [comment]
I apologize for the series of comments and the slight disorganization of the issue; I can modify the script with the reprex package in mind if needed. The script is quite simple and simply checks for existing files and overwrites them if necessary after aggregating some downloaded TIF format rasters from worldpop.org.

**Comment by bryce-carson** [comment]
Running the script serially, with `%do%`, somehow permitted the script to run to completion. I did also enable the `cores` argument in aggregate.

**Comment by rhijmans** [comment]
The overwrite error may have been fixed with https://github.com/rspatial/terra/issues/1522.  If not, it would be helpful to have a *minimal* and self-contained example.

**Comment by bryce-carson** [comment]
> The overwrite error may have been fixed with #1522. If not, it would be helpful to have a _minimal_ and self-contained example.

I will try to reproduce the error with the latest development version.
